tests/test_client_session.py

```diff
@@ -780,8 +780,9 @@ async def test_client_session_timeout_default_args(loop: Any) -> None:
 
 
 async def test_client_session_timeout_argument() -> None:
```

Comment:
Although, this test shouldn't exist in that case...
Especially on master.

Reply:
I do not think it is a good idea to introduce (semi-)breaking changes in a single PR

Reply:
Everything goes through the request() method, so if it's not currently working, then it never worked.
There's backwards-compatibility with float already on 3.x, but appears to only be when passed in the request() method, rather than on the ClientSession...
But, it's clearly not working on the ClientSession, so there's no risk of breakage here. We don't want to introduce a deprecated feature.

Reply:
removed unneeded part

Reply:
Test needs to be removed though, as it will fail now. The new test replaces this one.

Reply:
done